src/pages/workspaces/workspace/workflows/ExportWorkflowModal.tsx

```diff
@@ -0,0 +1,209 @@
+  const [error, setError] = useState<any>(undefined);
+
+  // whether the export operation is currently being performed and the spinner
+  // should be shown
+  const [exporting, setExporting] = useState<boolean>(false);
+
+  const [showingPostExportModal, setShowingPostExportModal] = useState<boolean>(false);
+
+  const { workspaces } = useWorkspaces();
+
+  const destinationWorkspaceSelectorId = useUniqueId();
+  const workflowNameInputId = useUniqueId();
+
+  // Helpers
+  const selectedWorkspace: WorkspaceInfo | undefined = _.find(
+    { workspace: { workspaceId: selectedWorkspaceId } },
+    workspaces
+  )?.workspace;
+
+  const doExport = async () => {
+    if (selectedWorkspace === undefined) {
+      setError('Cannot find destination workspace');
+    } else {
+      try {
+        setExporting(true);
+        await exportProvider.export(selectedWorkspace, workflowName);
+        onSuccess?.();
+
+        if (onGoToExportedWorkflow) {
+          setShowingPostExportModal(true);
+        }
+      } catch (error) {
+        setError(error instanceof Response ? await error.text() : error);
+        setExporting(false);
+      }
+    }
+  };
+
+  // Render helpers
+  const renderExportForm = () => {
+    const errors = validate(
+      { selectedWorkspaceId, workflowName },
+      {
+        selectedWorkspaceId: { presence: true },
+        workflowName: workflowNameValidation(),
+      }
+    );
+
+    const okButton = (
+      <ButtonPrimary tooltip={Utils.summarizeErrors(errors)} disabled={!!errors} onClick={doExport}>
+        {exportButtonText}
+      </ButtonPrimary>
+    );
+
+    return (
+      <Modal title={title} onDismiss={onDismiss} okButton={okButton}>
+        {!presetDestination && (
+          <>
+            <FormLabel htmlFor={destinationWorkspaceSelectorId} required>
+              Destination
+            </FormLabel>
+            <WorkspaceSelector
+              id={destinationWorkspaceSelectorId}
+              workspaces={_.filter(destinationWorkspace, workspaces)}
+              value={selectedWorkspaceId}
+              onChange={setSelectedWorkspaceId}
+              aria-label={undefined}
+            />
+          </>
+        )}
+        <FormLabel htmlFor={workflowNameInputId} required>
+          Name
+        </FormLabel>
+        <ValidatedInput
+          error={Utils.summarizeErrors(errors?.workflowName)}
+          inputProps={{ id: workflowNameInputId, value: workflowName, onChange: setWorkflowName }}
+        />
+        {exporting && spinnerOverlay}
+        {error && <ErrorView error={error} />}
+      </Modal>
+    );
+  };
+
+  const renderPostExport = () => {
+    // Note: selectedWorkspace cannot be undefined because if it were, the
+    // export would have failed and this modal would not be able to appear;
+    // onGoToExportedWorkflow cannot be undefined because if it were,
+    // exported could not have been set to true and this modal would not be
+    // able to appear
```

Comment:
Consider eliminating double negatives, negatives in general, and hypothetical past tense.

> This modal only appears after a successful export, so export-prerequisite `selectedWorkspace` must be defined.

Reply:
Updated comment for clarity.